op-supervisor/supervisor/backend/l1access/l1_accessor.go

```diff
@@ -159,8 +162,33 @@ func (p *L1Accessor) onFinalized(ctx context.Context, ref eth.L1BlockRef) {
 }
 
 func (p *L1Accessor) onLatest(ctx context.Context, ref eth.L1BlockRef) {
+	// Stop if the block is the same or older than the tip
+	if ref.Number == p.tipHeight && ref.Hash == p.tipHash {
+		p.log.Warn("L1 block is the same as the tip", "ref", ref)
+		return
+	}
+	if ref.Number < p.tipHeight {
+		p.log.Warn("L1 block is older than the tip", "ref", ref)
+		return
+	}
+
+	// Update the tip
 	p.tipHeight = ref.Number
+	p.tipHash = ref.Hash
 	p.log.Info("Updated latest known L1 block", "ref", ref)
+
+	// Check if the block is the next one and if it is, check for a reorg
+	if ref.Number != p.tipHeight+1 {
```

Comment:
You set `p.tip` to be `ref` here, but then just below we check `if ref.Number != p.tip.Number+1`. Doesn't that mean this check will always trigger?